glTF-BinExporter/RhinoDocGltfConverter.cs

```diff
@@ -176,18 +183,36 @@ int[] GetMaterials(RenderMaterial[] materials, RhinoObject rhinoObject)
                     return null;
                 }
 
+                Guid materialId;
                 if (material == null && options.UseDisplayColorForUnsetMaterials)
                 {
-                    Color4f objectColor = GetObjectColor(rhinoObject);
-                    materialIndices[i] = CreateSolidColorMaterial(objectColor, GetObjectName(rhinoObject));
+                    if (options.ExportLayers)
+                    {
+                        if (subObjects[i].Attributes.ColorSource == ObjectColorSource.ColorFromLayer)
+                        {
+                            materialIndices[i] = GetLayerMaterial(rhinoObject);
+                            continue;
```

Comment:
In this context these continues here and on line 205 are confusing. Much of this loop needs to be made more coherent. It's hard to follow which conditions will set which material index.